api/base/serializers.py

```diff
@@ -343,6 +343,25 @@ def resolve(self, resource):
             )
         )
 
+    def process_related_counts_parameters(self, params):
+        """
+        Processes related_counts parameter.
+
+        Can either be a True/False value for fetching counts on all fields, or a comma-separated list for specifying
+        individual fields.  Ensures field for which we are requesting counts is a relationship field.
+        """
+        if utils.is_truthy(params) or utils.is_falsy(params):
+            return params
+
+        field_counts_requested = [val for val in params.split(',')]
+        for count_field in field_counts_requested:
+            if count_field not in [field for field in self.parent.fields if getattr(self.parent.fields[field], 'json_api_link', False)]:
```

Comment:
You can move the list-comp outside of the loop so that it isn't computed more than once. Also, you can make it a set comprehension in order to get constant-time membership checks.